packages/staking/src/features/store/delegationPortfolioStore/useDelegationPortfolioStore.ts

```diff
@@ -84,6 +87,51 @@ export const useDelegationPortfolioStore = create(
           state.pendingSelectedPortfolio = undefined;
           state.viewedStakePool = undefined;
         }),
+      hydrate: ({
+        poolIds,
+        stakePoolProvider,
+        view,
+        poolsView,
+      }: {
+        poolIds: Wallet.Cardano.PoolId[];
+        stakePoolProvider: Wallet.StakePoolProvider;
+        view: 'popup' | 'expanded';
+        poolsView: BrowsePoolsView;
+      }) => {
+        if (poolIds.length === 0) {
+          set((state) => {
+            Object.assign(state, {
+              browsePoolsView: poolsView,
+              hydrated: true,
+              view,
+            });
+          });
+
+          return;
+        }
+
+        getPoolInfos({
+          poolIds,
+          preserveOrder: true,
+          stakePoolProvider,
+          status: [
+            Wallet.Cardano.StakePoolStatus.Activating,
+            Wallet.Cardano.StakePoolStatus.Active,
+            Wallet.Cardano.StakePoolStatus.Retiring,
+          ],
+        })
+          .then((selectedStakePools) => {
+            set((state) => {
+              Object.assign(state, {
+                ...atomicStateMutators.selectPools({ stakePools: selectedStakePools, state }),
+                browsePoolsView: poolsView,
+                hydrated: true,
+                view,
+              });
+            });
+          })
+          .catch(() => null);
```

Comment:
What will happen with `view` and other properties when it fails? We also may experience an infinite loop due to `hydrated` being always `false`.

Note: I think it's possible to have async actions in Zustand but if not then you can always set the state in `catch`.

```suggestion
        const selectedStakePools = await getPoolInfos({
          poolIds,
          preserveOrder: true,
          stakePoolProvider,
          status: [
            Wallet.Cardano.StakePoolStatus.Activating,
            Wallet.Cardano.StakePoolStatus.Active,
            Wallet.Cardano.StakePoolStatus.Retiring,
          ],
        }).catch(() => []);
        
        set((state) => {
              Object.assign(state, {
                ...atomicStateMutators.selectPools({ stakePools: selectedStakePools, state }),
                browsePoolsView: poolsView,
                hydrated: true,
                view,
              });
            });
```

Reply:
I've change this so we have empty array for browsePoolsView and hydrated in case of failed request